README.rst

```diff
@@ -436,6 +451,28 @@ For manual control over positioning (e.g. for multi-threaded use),
 you may specify `position=n` where `n=0` for the outermost bar,
 `n=1` for the next, and so on.
 
+Writing messages
+~~~~~~~~~~~~~~~~~~~~
+Since ``tqdm`` uses a simple printing mechanism to display progress bars,
+you should not write any message in the terminal using ``print()``.
+
+To write messages in the terminal without any collision with ``tqdm`` bar
+display, a ``.write()`` method is provided:
+
+.. code:: python
+
+    from tqdm import tqdm, trange
+    from time import sleep
+
+    bar = trange(10)
+    for i in bar:
+        # Print using tqdm class method .write()
+        tqdm.write("Done task %i" % i)
+        # Can also use instance method bar.write()
```

Comment:
But that's obvious if it's a class method
On 22 Apr 2016 7:08 pm, "Stephen L." <notifications@github.com> wrote:

> In README.rst <https://github.com/tqdm/tqdm/pull/150#discussion_r60779668>
> :
>
> > +Since ``tqdm`` uses a simple printing mechanism to display progress bars,
> > +you should not write any message in the terminal using ``print()``.
> > +
> > +To write messages in the terminal without any collision with ``tqdm`` bar
> > +display, a ``.write()`` method is provided:
> > +
> > +.. code:: python
> > +
> > +    from tqdm import tqdm, trange
> > +    from time import sleep
> > +
> > +    bar = trange(10)
> > +    for i in bar:
> > +        # Print using tqdm class method .write()
> > +        tqdm.write("Done task %i" % i)
> > +        # Can also use instance method bar.write()
>
> @graingert <https://github.com/graingert> You're right but I wanted to
> make things simple to understand for the reader, and also to get a shorter
> comment (flake8!). But yeah, the goal was just to point that .write() could
> be used both from tqdm class or from an instance of tqdm.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/tqdm/tqdm/pull/150/files/8e1ce8bf117b8b2b7078390aee7488fcd34d053a#r60779668>
>